File: Unidad 4-Arreglos/ejercicio8.py
import random,math

#Autor: Jose Eduardo Mendez Verdejo    09/Marzo/19
#Entradas: Un vector de tamano "n" con numeros random entre 0,100
#Salidas: la media, la varianza y la desviacion estandar
#Proceso: Calcular la media cuando se agregan elementos al vector sumando el elemento, y despues la suma total
#		  se divide entre le numero de elementos, la varianza se calcula aproximadamente igual que la media, solo
#		  que se suma (elemento - media) y se divide entre el total de elementos, la desviacion es la raiz de la varianza

#----- Genera los numeros random del vector -----
def generateArray(array,media):
	for i in range(0,n):
		array[i] = random.randrange(0,100)

		#se suman los numeros para calcular la media
		media = media + array[i]
	
	#se divine entre el numero de elementos del vector
	media = media / n

	return media	

#----- Se obtiene la varianza -----
def getVariazna(array,arrayVarianza,media):

	varianza = 0
	j = 0

	for i in array:

		#a un vector nuevo se le otorga el resultado de (x-media)^2
		arrayVarianza[j] = (i - media) * (i - media) 
		varianza += arrayVarianza[j] #se suman los elementos

		j += 1

	varianza /= n #se divide entre n y se obtiene la varianza
	return varianza

#----- Obtener desviacion estandar -----
def getDesEst(varianza):
	desviacionEstandar = math.sqrt(varianza) #se obtiene la raiz de varianza
	return desviacionEstandar

#se define el tamaño del vector
n = int(input("Define le tamano del vector: "))

array = [None] * n	
arrayVarianza = [None] * n
media = 0

media  = generateArray(array,media)
varianza = getVariazna(array,arrayVarianza,media)
desviacionEstandar = getDesEst(varianza)

print("Vector ",array)
print("media: ",media)
print("varianza: ",varianza)
print("desviacion Estandar: ",desviacionEstandar)

File: Unidad 4-Arreglos/Ejercicio8.js
/*
	Autor: Carlos Chan Gongora
	Entradas: El tamaño del vector.
	Salidas: La media, desviacion estandar y varianza.
	Procedimiento general: Lee el largo del vector y posterior lo rellena con numeros aleatorios entre 0 y 100,
	luego calcula la media aritmetica, la varianza y la desviacion.
*/

// Entradas
let vector = new Array();
let tam = 0;
let media = 0, 
	desviacion = 0, 
	varianza = 0;
let resultados = "";

tam = parseInt(prompt("Ingresa el largo del vector: "));

// Proceso
rellenarVector(vector, tam);
media = calcularMedia(vector, tam);
varianza = calcularVarianza(vector, tam);
desviacion = calcularDesviacion(varianza);
	
// Salidas
resultados = "Media: " + media + "\n" + "Desviacion estandar: " + desviacion + "\n" + "Varianza de una muestra: " + varianza;
alert(resultados);
imprimirVector(vector, tam);

function rellenarVector(vec, tam){
	for(let i = 0; i < tam; i++){
		vec[i] = Math.round(Math.random() * 100);
	}
}

function calcularMedia(vec, tam){
	let media = 0;
	for(let i = 0; i < tam; i++){
		media += vec[i];
	}
	media = media / tam;
	return media;
}

function calcularPromedio(vec, tam){
  let suma = 0;
  for (let i = 0; i < tam; i++) {
    suma += vec[i];
  }
  suma = suma / tam;
  return suma;
}

function calcularVarianza(vec, tam){
  let promedio = calcularPromedio(vec, tam);
  let varianza = 0;
  for (let i = 0; i < tam; i++) {
    varianza += Math.pow((vec[i] - promedio), 2);
  }
  varianza = varianza / (tam - 1);
  return varianza;
}

function calcularDesviacion(varianza){
  return Math.sqrt(varianza);
}

function imprimirVector(vec, tam){
	let impresion = "";
	for(let i = 0; i < tam; i++){
		impresion += vec[i] + ", ";
	}
	alert("Vector: " + impresion);
}